frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -87,302 +49,116 @@ const getColumns = () => {
   ];
 };
 
-const hasTaints = (node: NodeKind) => {
-  return !_.isEmpty(node.spec.taints);
-};
+const getRows = ({ componentProps, customData }): NodeTableRow[] => {
+  const { data } = componentProps;
+  const { selectedNodes, setSelectedNodes, setVisibleRows, visibleRows } = customData;
 
-const hasOCSTaint = (node: NodeKind) => {
-  const taints: Taint[] = node.spec.taints || [];
-  return taints.some((taint: Taint) => _.isEqual(taint, ocsTaint));
-};
+  const isSelected = (selected: NodeKind[], nodeUID: string) =>
+    selected.map((node) => node.metadata.uid).includes(nodeUID);
+
+  const filteredData = data.filter((node: NodeKind) => hasOCSTaint(node) || !hasTaints(node));
```

Comment:
You can consider moving these outside. `getRows` is the hotspot, and these functions will always be created when its called.